packages/datastore-storage-adapter/src/ExpoSQLiteAdapter/ExpoSQLiteDatabase.ts

```diff
@@ -0,0 +1,283 @@
+				tx.executeSql(
+					statement,
+					params,
+					() => resolve(),
+					(_, error) => {
+						reject(error);
+						return true;
+					}
+				);
+			});
+		});
+	}
+
+	public async batchQuery(
+		queryStatements: Set<ParameterizedStatement>
+	): Promise<any[]> {
+		const results = [];
+		await new Promise((resolve, reject) => {
+			this.db.readTransaction(function (tx) {
+				for (const [statement, params] of queryStatements) {
+					tx.executeSql(
+						statement,
+						params,
+						function (_tx, res) {
+							results.push(res.rows._array[0]);
+						},
+						(_, error) => {
+							reject(error);
+							return true;
+						}
+					);
+				}
+				resolve();
+			});
+		});
+		return results;
+	}
+
+	public async batchSave(
+		saveStatements: Set<ParameterizedStatement>,
+		deleteStatements?: Set<ParameterizedStatement>
+	): Promise<void> {
+		await new Promise((resolve, reject) => {
+			this.db.transaction(function (tx) {
+				for (const [statement, params] of saveStatements) {
+					tx.executeSql(
+						statement,
+						params,
+						() => {},
+						(_, error) => {
+							reject(error);
+							return true;
+						}
+					);
+				}
+
+				if (deleteStatements) {
+					for (const [statement, params] of deleteStatements) {
+						tx.executeSql(
+							statement,
+							params,
+							() => {},
+							(_, error) => {
+								reject(error);
+								return true;
+							}
+						);
+					}
+				}
+				resolve();
+			});
+		});
+	}
+
+	public async selectAndDelete(
+		query: ParameterizedStatement,
+		_delete: ParameterizedStatement
+	): Promise<any[]> {
+		let results = [];
+
+		const [queryStatement, queryParams] = query;
+		const [deleteStatement, deleteParams] = _delete;
+
+		results = await new Promise((resolve, reject) => {
+			this.db.transaction(function (tx) {
+				let tempresults;
+				tx.executeSql(
+					queryStatement,
+					queryParams,
+					(_tx, res) => {
+						tempresults = res.rows._array;
+					},
+					(_, error) => {
+						reject(error);
+						return true;
+					}
+				);
+				tx.executeSql(
+					deleteStatement,
+					deleteParams,
+					() => {},
+					(_, error) => {
+						reject(error);
+						return true;
+					}
+				);
+				resolve(tempresults);
+			});
+		});
+		return results || [];
+	}
+
+	private async executeStatements(statements: string[]): Promise<void> {
+		await new Promise((resolve, reject) => {
+			this.db.transaction(function (tx) {
+				for (const statement of statements) {
+					tx.executeSql(
+						statement,
+						[],
+						() => {},
+						(_, error) => {
+							reject(error);
+							return true;
+						}
+					);
+				}
+				resolve();
+			});
+		});
+	}
+
+	private async closeDB() {
+		if (this.db) {
+			logger.debug('Closing Database');
+			// closing database is not supported by expo-sqlite.
+			// Workaround is to access the private db variable and call the close() method.
+			const tempdb: any = this.db;
+			await tempdb._db.close();
```

Comment:
Fwiw, we are already doing a somewhat unsafe cast regardless we may as well just opt for readability and keep the `any`. (I think the proposed casting would have to tack on the `_db` property too, making it even more unclear)

I agree we shouldn't need a tempdb though:
```suggestion
			await (this.db as any)._db.close();
```

Reply:
I'll use this instead of a temp variable.